src/main/java/ch/admin/bar/siard2/jdbc/PostgresMetaColumns.java

```diff
@@ -203,19 +203,60 @@ private String getTypeName(String sTypeName, int iType)
           rs.close();
         }
         ***/
+        String sSchemaName = this.getString(2); // TABLE_SCHEM
+        String sTableName = this.getString(3); // TABLE_NAME
+        String sColumnName = this.getString(4); // COLUMN_NAME
+
         if (PostgresType.INTERVAL.getKeyword().equals(sTypeName)) 
         {
-          String sSchemaName = this.getString(2);
-          String sTableName = this.getString(3);
-          String sColumnName = this.getString(4);
           sTypeName = getIntervalTypeName(_conn, sSchemaName, sTableName, sColumnName, sTypeName);
         }
+
+        //add type length/precision
+        sTypeName = addTypeLengthAndPrecision(sTypeName, iType, getAttTypMod(_conn, sSchemaName, sTableName, sColumnName));
       }
       catch(ParseException pe) { throw new SQLException("Parsing of "+sTypeName+" failed ("+EU.getExceptionMessage(pe)+")!"); }
     }
     return sTypeName;
   } /* getTypeName */
-  
+
+  private String addTypeLengthAndPrecision(String sTypeName, int iType, int atttypmod) {
+    if (atttypmod < 0){
+      return sTypeName;
+    }
```

Comment:
you could just put it on a single line:

`if (atttypmod < 0) return sTypeName;`
Guard clauses used like this greatly improve readability of code